tvb_framework/tvb/adapters/creators/siibra_base.py

```diff
@@ -0,0 +1,463 @@
+                maskdata = np.asanyarray(childmask.dataobj)
+            else:
+                assert (childmask.affine == affine).all()
+                maskdata = np.maximum(maskdata, np.asanyarray(childmask.dataobj))
+        else:
+            # we get here only if the for loop was not interrupted by 'break'
+            if maskdata is not None:
+                return Nifti1Image(maskdata, affine)
+
+    if mask is None:
+        # No map of the requested type found for the region.
+        LOGGER.warn(
+            f"Could not compute {maptype.name.lower()} mask for {region.name} in {spaceobj.name}."
+        )
+        if consider_other_types:
+            for other_maptype in (set(siibra.MapType) - {maptype}):
+                mask = region.build_mask(
+                    space, resolution_mm, other_maptype, threshold_continuous, consider_other_types=False
+                )
+                if mask is not None:
+                    LOGGER.info(
+                        f"A mask was generated from map type {other_maptype.name.lower()} instead."
+                    )
+                    return mask
+        return None
+
+    if (threshold_continuous is not None) and (maptype == siibra.MapType.CONTINUOUS):
+        data = np.asanyarray(mask.dataobj) > threshold_continuous
+        LOGGER.info(f"Mask built using a continuous map thresholded at {threshold_continuous}.")
+        assert np.any(data > 0)
+        mask = Nifti1Image(data.astype("uint8").squeeze(), mask.affine)
+
+    return mask
+
+
+# ######################################## COMMON CONNECTIVITY METHODS #################################################
+def get_connectivity_component(parcellation, component):
+    # type: (str, Component2Modality) -> []
+    """
+        :return: a list of all available connectivity components (weights/tract lengths)
+    """
+    modality = component.value
+    all_conns = siibra.get_features(parcellation, modality)
+
+    if len(all_conns) == 0:
+        LOGGER.error(f'No connectivity {component} were found in parcellation {parcellation}!')
+        return None
+    return all_conns
+
+
+def get_hemispheres_for_regions(region_names):
+    """ Given a list of region names, compute the hemispheres to which they belon to """
+    LOGGER.info(f'Computing hemispheres for regions')
+    hemi = []
+    for name in region_names:
+        if 'right' in name:
+            hemi.append(1)
+        # TODO: regions referring to both hemispheres are put in the left hemisphere; change this?
+        else:
+            hemi.append(0)
+
+    return hemi
+
+
+def get_regions_positions(regions):
+    """ Given a list of regions, compute the positions of their centroids """
+    LOGGER.info(f'Computing positions for regions')
+    positions = []
+
+    space = siibra.spaces.MNI152_2009C_NONL_ASYM  # commonly used space in other examples
+
+    for r in regions:
+        # get centroids list
+        centroids = compute_centroids(r, space)
```

Comment:
Maybe here adjust the `compute_centroids` recent function, to return just the first value (as a tuple) instead of returning more and taking just the first one ?

Reply:
Sure, will do so